### ISSUE TITLE
"Loading Complete" message interrupts the screen reader & the rest of the page is not read.

### ISSUE THREAD

**Issue by misupea** [nvaccess/nvda#17986]
### Steps to reproduce:
1. Start  the latest version of NVDA (2024.4.2)
2. Ensure "Say all on page load" is enabled in the Browse settings
3. Open Edge Browser
4. Navigate to a website such as: https://hydev.powerappsportals.com/en-US/
5. Observe the behavior 
6. Navigate to NVDA's homepage: https://www.nvaccess.org/download/
7. Observe the behavior
### Actual behavior:

On (https://www.nvaccess.org/download/) the screen reader starts to read the page & is interrupted with a "Loading complete" message. The rest of the page is not read. 

<img width="259" alt="Image" src="https://github.com/user-attachments/assets/23203d5b-89a3-413e-be63-fa0f015d3767" />

On the NVDA homepage, the page starts to read as normal and is interrupted with the "Loading Complete" message. The screen reader stops reading the page. 
<img width="244" alt="Image" src="https://github.com/user-attachments/assets/f44e00b1-bf2a-49ce-a52f-12a404e57ccd" />

This behavior is inconsistent with different websites, but the majority of them are interrupted with the page load complete message. Sometimes the message is not announced at all and the page reads as expected. 
### Expected behavior:

After the "Loading Complete" announcement, the screen should be read as usual and Say all on page load should work. 
#### NVDA installed/portable/running from source: 
Device: Surface Laptop Studio

#### NVDA version:
Latest Version - 2024.4.2

#### Windows version:
Latest version - 

![Image](https://github.com/user-attachments/assets/bfc56876-c830-4850-953c-9fbbb811b600)

#### Name and version of other software in use when reproducing the issue:
Edge Browser

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes. Version 2024.4 does not encounter this issue. This issue happened after installing the newest version. 

#### If NVDA add-ons are disabled, is your problem still occurring?
Yes

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes. The issue happens across devices and is present with my entire team.

**Comment by josephsl** [comment]
Hi,

Cause: UIA notification event raised by Edge prevents say all on page load from completing. A while ago I was thinking about best ways to resolve this issue but put it aside due to life events.

Thanks.

**Comment by misupea** [comment]
> Hi,
> 
> Cause: UIA notification event raised by Edge prevents say all on page load from completing. A while ago I was thinking about best ways to resolve this issue but put it aside due to life events.
> 
> Thanks.
From a developer standpoint, is there any way to work around this?

**Comment by josephsl** [comment]
Hi,

This will involve editing UIA notification event handler defined in base UIA NVDA object to handle active say all (like the issue described here), meaning that this is something that should be resolved from NVDA. I won't be able to work on it until mid-May unless someone wants to take this on.

Thanks.

**Comment by pbsinnett** [comment]
There's an excellent add-on that suppresses Edge announcements which works with NVDA version 2024.4.2, but it will stop working when 2025.1 is released. It hasn't been updated in a while, but here it is if anyone wants to check it out.

https://github.com/addonFactory/MSEdgeDiscardAnnouncements

**Comment by josephsl** [comment]
Hi,

Coming back to this issue...

As part of a collection of UIA notification event handler changes, I plan to work on resolving this issue in a pull request targeting 2025.2.

Thanks.

**Comment by austin-reed-iseatz** [comment]
Hello, I am currently facing an issue that may be related to this. Currently on version 2025.1 using Google Chrome on the latest version and Windows 11, when on a webpage that uses an interstitial while loading content with aria-live="assertive" for the interstitial's content, the interstitial's content will begin to be read out. When the content that was loading is done loading, the interstitial goes away along with the html elements that contained the content that was being read out by the screen reader and new content is displayed and an alert message plays letting a message that lets the user know that the page is done loading is announced. Once that alert is done, the interstitial content that was never fully read out start being read again even though it is no longer on the page. Is there any kind of setting configuration that could prevent this?

**Comment by josephsl** [comment]
Hi,

On the surface, yes, it might be related. However, the underlying mechanism might be different - the one I addressed comes from UIA notification handling during say all, whereas the one you are describing may have to do with an alert or ive region event.

Thakns.
